obspy/io/xseed/tests/test_parser.py

```diff
@@ -533,6 +533,110 @@ def test_create_resp_from_xseed(self):
+        for filename in (sts2_resp_file, rt130_resp_file):
+            with open(filename, 'rt') as f:
+                data = f.read().encode('utf-8')
+            p = Parser(data)
+            p.get_resp()
+
+    def test_resp_round_trip(self):
+        single_seed = os.path.join(
+            self.path,
+            '../../../../',
+            'core/tests/data/IRIS_single_channel_with_response.seed')
+        # Make parser and get resp from SEED
+        seed_p = Parser(single_seed)
+        resp_from_seed = seed_p.get_resp()[0][1]
+        resp_from_seed.seek(0)
+        resp_from_seed = resp_from_seed.read()
+
+        # make parser from resp made above and make a resp from it
+        resp_p = Parser(resp_from_seed)
+        resp_from_resp = resp_p.get_resp()[0][1]
+        resp_from_resp.seek(0)
+        resp_from_resp = resp_from_resp.read()
+
+        # Strip out string constants that differ.
+        ignored_fields = ['B054F05', 'B054F06', 'B053F05', 'B053F06']
+        resp_from_resp = [line for line in resp_from_resp.split('\n')
+                          if line[:7] not in ignored_fields ]
+        resp_from_seed = [line for line in resp_from_seed.split('\n')
+                          if line[:7] not in ignored_fields ]
+        #compare
+        self.assertEqual(resp_from_resp, resp_from_seed)
+
+
+    def test_parse_resp(self):
+        """
+        Tests parsing a RESP file by calling Parser._parse_resp(string)
+        """
+        sts2_resp_file = os.path.join(self.path,
+                                      'RESP.XX.NS085..BHZ.STS2_gen3.120.1500')
+        rt130_resp_file = os.path.join(self.path,
+                                       'RESP.XX.NR008..HHZ.130.1.100')
+        for filename in (sts2_resp_file, rt130_resp_file):
+            with open(filename, 'rt') as f:
+                data = f.read().encode('utf-8')
+            p = Parser()
+            p._parse_resp(data)
+            p.get_resp()
+
+    def test_join_sensor_dl_resps(self):
+        """
+        Test creation of response from sensor and datalogger.
+        """
+        sts2_resp_file = os.path.join(self.path,
+                                      'RESP.XX.NS085..BHZ.STS2_gen3.120.1500')
+        rt130_resp_file = os.path.join(self.path,
+                                       'RESP.XX.NR008..HHZ.130.1.100')
+        sensor = Parser(sts2_resp_file)
+        dl = Parser(rt130_resp_file)
+        combined = Parser.combine_sensor_dl_resps(sensor=sensor, datalogger=dl)
+
+        # Try to use a sensor and dl from bw seed files.
+        sensor = Parser(self.BW_SEED_files[0])
+        dl = Parser(self.BW_SEED_files[1])
+        combined = Parser.combine_sensor_dl_resps(sensor=sensor, datalogger=dl)
+        # combined.write_resp('.')
+
+    def test_join_wrong_types(self):
+        with self.assertRaises(TypeError) as e:
+            Parser.combine_sensor_dl_resps(None, None)
+
+    # def test_read_resp_with_b61(self):
```

Comment:
I think we want to assert that an error is raised, as that is what we want to test. No?
`expectedFailure` seems more for this test may fail, but don't mark as failed test.
I'll change if not.